app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,330 @@
+      };
+
+      var deregisterRootScopeWatches = function() {
+        _.each(rootScopeWatches, function(deregister) {
+          deregister();
+        });
+        rootScopeWatches = [];
+      };
+
+      var hideIfNoProject = function(projectName) {
+        if(!projectName) {
+          drawer.drawerHidden = true;
+        }
+      };
+
+      var render = function() {
+        $rootScope.$evalAsync(function () {
+          countUnreadNotificationsForAllGroups();
+          // NOTE: we are currently only showing one project in the drawer at a
+          // time. If we go back to multiple projects, we can eliminate the filter here
+          // and just pass the whole array as notificationGroups.
+          // if we do, we will have to handle group.open to keep track of what the
+          // user is viewing at the time & indicate to the user that the non-active
+          // project is "asleep"/not being watched.
+          drawer.notificationGroups = _.filter(notificationGroups, function(group) {
+            return group.project.metadata.name === $routeParams.project;
+          });
+        });
+      };
+
+      // TODO: follow-on PR to decide which of these events to toast,
+      // via config in constants.js
+      var eventWatchCallback = function(eventData) {
+        eventsByNameData = eventData.by('metadata.name');
+        eventsMap = formatAndFilterEvents(eventsByNameData);
+        // TODO: Update to an intermediate map, so that we can then combine both
+        // events + notifications into the final notificationGroups output
+        notificationGroups = sortNotificationGroups(eventsMap);
+        render();
+      };
+
+      // TODO: Follow-on PR to update & add the internal notifications to the
+      // var notificationWatchCallback = function(event, notification) {
+      //   // will need to add .event = {} and immitate structure
+      //   if(!notification.lastTimestamp) {
+      //     // creates a timestamp that matches event format: 2017-08-09T19:55:35Z
+      //     notification.lastTimestamp = moment.parseZone(new Date()).utc().format();
+      //   }
+      //   clientGeneratedNotifications.push(notification);
+      // };
+
+      var iconClassByEventSeverity = {
+        Normal: 'pficon pficon-info',
+        Warning: 'pficon pficon-warning-triangle-o'
+      };
+
+      angular.extend(drawer, {
+        drawerHidden: true,
+        allowExpand: true,
+        drawerExpanded: false,
+        drawerTitle: 'Notifications',
+        hasUnread: false,
+        showClearAll: true,
+        showMarkAllRead: true,
+        onClose: function() {
+          drawer.drawerHidden = true;
+        },
+        onMarkAllRead: function(group) {
+          _.each(group.notifications, function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+          });
+          render();
+          $rootScope.$emit('notification-drawer:mark-read');
```

Comment:
Def think we should be pretty proactive about consistent naming convention for these events, this is good.